tozny/resource_identity.go

```diff
@@ -0,0 +1,157 @@
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"email": {
+				Description: "The email address associated with this user.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"client_registration_token": {
+				Description: "A registration token for the realm allowed to create identities",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"broker_target_url": {
+				Description: "The base link for password resets",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"password": {
+				Description: "The password for this identity. Ideally this comes from a secret store of some kind.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"first_name": {
+				Description: "The first name associated with this identity",
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"last_name": {
+				Description: "The last name associated with this identity",
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"recovery_email_ttl": {
+				Description: "The length of time a recovery email is valid for",
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     60,
+				ForceNew:    true,
+			},
+		},
+	}
+}
+
+func resourceIdentityCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	toznySDK, err := MakeToznySDK(d, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	realmName := d.Get("realm_name").(string)
+	registrationToken := d.Get("client_registration_token").(string)
+	username := d.Get("username").(string)
+	email := d.Get("email").(string)
+	password := d.Get("password").(string)
+	brokerTargetURL := d.Get("broker_target_url").(string)
+	emailExpiryMinutes := d.Get("recovery_email_ttl").(int)
+	realm := e3db.Realm{
+		Name:               realmName,
+		App:                "account",
+		APIEndpoint:        toznySDK.APIEndpoint,
+		BrokerTargetURL:    brokerTargetURL,
+		EmailExpiryMinutes: emailExpiryMinutes,
+	}
+
+	identity, err := realm.Register(username, password, registrationToken, email, "", "")
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(identity.ClientID)
+
+	return diags
+}
+
+func resourceIdentityRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	// There is nothing to really update for an identity at this time, so implement as a noop.
+	return diags
+}
+
+func resourceIdentityDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	toznySDK, err := MakeToznySDK(d, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	realmName := strings.ToLower(d.Get("realm_name").(string))
```

Comment:
can we move this lowercasing to e3db-go?

Reply:
Will bring this up at standup. This is an e3db-clients-go method, and uses the realm domain. In doing this, I've become a bit confused as to what is the chicken and the egg with domain/name in realms... and we may be sometimes inefficiently requesting realm info at times. We should get aligned before I add logic like this to a more shared library.